### DISCUSSION TITLE
kotlin怎么将文字和本地图片合并成一条消息发送

### DISCUSSION THREAD

**Discussion by tracker93** [mamoe/mirai#821] [Q&A] [answered]
`var ins: InputStream = FileInputStream("image/V[4YC)[`]P%)7[@I0I~ODV0.png")
`friend.sendImage(ins)
`friend.sendMessage(PlainText(msg).plus(""))

如上，我无法在文字消息后添加一个图片。或者有没有办法在文字消息后面添加一个收藏的表情图片一起合并发送

**Comment by Him188** [comment, marked as answer]
```kotlin
val image = friend.uploadImage(inputStream)
friend.sendImage(PlainText("") + image)
```

**Reply by tracker93** [reply]
哇，好高效。谢谢

**Reply by tracker93** [reply]
`friend.sendMessage(PlainText(msg) + Image(image.imageId))`
是需要这样子使用，直接+image会输出一个图片id.
但是这个方法如果我使用group上传合并的图文消息后，终端打印了机器人的消息，但是实际上群里没有消息是什么情况？有时候机器人发送网址也会这样子，这个是tx的机制，还是mirai上传比较慢导致的？

**Reply by tracker93** [reply]
有没有方法直接发送我收藏在服务器中的表情包？这种方式会不会可以减少图片上传很慢的问题？

**Reply by sandtechnology** [reply]
这是腾讯服务器的风控，和mirai无关

**Reply by Him188** [reply]
在 2.0-RC 才支持群图片发给好友
